include/io.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2017 Glen Joseph Fernandes
+(glenjofe@gmail.com)
+
+Distributed under the Boost Software License, Version 1.0.
+(http://www.boost.org/LICENSE_1_0.txt)
+*/
+#ifndef BOOST_NUMERIC_IO_HPP
+#define BOOST_NUMERIC_IO_HPP
+
+#include <istream>
+#include <ostream>
+
+namespace boost {
+namespace numeric {
+namespace detail {
+
+template<class CharT, class Traits, class T>
+inline void
+output(std::basic_ostream<CharT, Traits>& os, const T& value)
+{
+    os << value;
+}
+
+template<class CharT, class Traits>
+inline void
+output(std::basic_ostream<CharT, Traits>& os, signed char value)
+{
+    os << static_cast<int>(value);
+}
+
+template<class CharT, class Traits>
+inline void
+output(std::basic_ostream<CharT, Traits>& os, unsigned char value)
+{
+    os << static_cast<unsigned int>(value);
+}
+
+template<class CharT, class Traits, class T>
+inline void
+input(std::basic_istream<CharT, Traits>& is, T& value)
+{
+    is >> value;
+}
+
+template<class CharT, class Traits>
+inline void
+input(std::basic_istream<CharT, Traits>& is, signed char& value)
+{
+    int result;
+    is >> result;
+    value = result;
```

Comment:
Good point; that new abstraction isn't the best idea for input. I updated the latter for parity with what he had before.